### ISSUE TITLE
Generalize exclude rules support for long lines for URLs (etc.)

### ISSUE THREAD

**Issue by timabbott** [zulip/zulint#6]
See https://github.com/zulip/zulip/pull/13078#issuecomment-525040589 for details

**Comment by gnprice** [comment]
On the other thread, Tim quoted this line from the linter:
```
# Don't throw errors for markdown format URLs
not re.search(r"^\[[ A-Za-z0-9_:,&()-]*\]: http.*", line) and
```

The line it failed to exempt was
```
[android7.0-tls-issue-so]: https://stackoverflow.com/questions/39133437/sslhandshakeexception-handshake-failed-on-android-n-7-0/42047877#42047877
```

I think the immediate character it's missing is `.`.

More generally, it'd be good to make the pattern agree with whatever exactly is the actual Markdown syntax for these named links. That way we avoid a continuing stream of small bugs like this one.

(/cc @andersk )
